manila/tests/scheduler/drivers/test_external.py

```diff
@@ -0,0 +1,275 @@
+# Copyright 2025 SAP SE or an SAP affiliate company.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+"""
+Tests for the external scheduler api call.
+"""
+
+from unittest.mock import MagicMock
+from unittest.mock import patch
+
+import jsonschema
+import requests
+
+from manila import context
+from manila.scheduler.drivers.external import call_external_scheduler_api
+from manila.scheduler.weighers.base import WeighedObject
+from manila.tests.scheduler.drivers import test_base
+from manila.tests.scheduler import fakes
+
+# The expected request schema for the external scheduler API.
+# It should contain the spec, hosts, and weights.
+REQUEST_SCHEMA = {
+    "type": "object",
+    "properties": {
+        "spec": {"type": "object"},
+        "context": {"type": "object"},
+        "hosts": {
+            "type": "array",
+            "items": {
+                "type": "object",
+                "properties": {
+                    "host": {"type": "string"},
+                },
+                "required": ["host"],
+            },
+        },
+        "weights": {"type": "object"},
+    },
+    "required": ["spec", "context", "hosts", "weights"],
+    "additionalProperties": False,
+}
+
+
+class ExternalSchedulerAPITestCase(test_base.SchedulerTestCase):
+    def setUp(self):
+        super(ExternalSchedulerAPITestCase, self).setUp()
+        self.flags(external_scheduler_api_url='http://127.0.0.1:1234')
+        self.flags(external_scheduler_timeout=10)
+        self.h1 = fakes.FakeHostState('host1', {'cell_uuid': 'i-dont-matter'})
```

Comment:
`cell_uuid` shows nova terminology that we don't have here - though it does not really matter ;)